src/composition.rs

```diff
@@ -735,10 +737,10 @@ impl Composition {
         let mut analyzer = CompositionQueryAnalyzer::default();
         analyzer.analyze(expr_statement.clone());
 
-        if analyzer.fields.contains(&field.to_string()) {
+        if analyzer.fields.contains(&field) {
```

Comment:
what are the ramifications of inverting where these strings are stored?

Reply:
We save one additional allocation per String.

The strings are received in the LspServer as a String object. The downstream mechanisms require String objects. The function signatures for composition required a conversion to a string primitive (string slice), and then re-allocation as a String object. That was pointless; since we can just pass in the original String object (it's a move anyways -- borrow checker is fine). Save one additional allocation per string.